### PR TITLE
Add integration tests for Horizon

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -3604,6 +3605,42 @@ function oncontroller_run_tempest()
     return $tempestret
 }
 
+function oncontroller_run_integraion_test()
```
Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: typo integraion

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -3816,6 +3853,13 @@ function oncontroller_testsetup()
         tempestret=$?
     fi
 
+    # Run Horizon integration test if configure to do so
+    integrationret=0
+    if [ "$want_integration_test" = "1" ]; then
+        # Horizon / Dashboard is installed in the controller node
+	onconroller_run_integration_test
```
Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: typo: onconroller

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: Done!

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: Thanks! Done!

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -3604,6 +3605,42 @@ function oncontroller_run_tempest()
...
+    safely zypper -n in python-selenium
+    safely zypper -n in python-nose
+    safely zypper -n in python-xvfbwrapper
+
+    # Configure the tests to be headless
+    export WITH_SELENIUM=1
+    export SELENIUM_HEADLESS=1
+    export INTEGRATION_TESTS=1
+
+    cd /srv/www/openstack-dashboard/openstack_dashboard/test/integration_tests
```
Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: Should we popd and pushd into the dir so we end up in the same place as when we started?

Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: Or maybe we dont even need to mv into that dir, instead telling crudini to do the changes to the $basepath/horizon.conf file instead?

**Comment by Itxaka** [comment]
take into account that running the integration tests for horizon takes a while:

Ran 68 tests in 2742.159s

So Im not sure if we have some kind of time limit on the test runner that would kill the CI job if it goes over X minutes. In that case, it should be changed to account for this tests I guess.

**Comment by nicolasbock** [comment]
Travis doesn't like your formatting :) But from my side

+1

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -3816,6 +3880,13 @@ function oncontroller_testsetup()
         tempestret=$?
     fi
 
+    # Run Horizon integration test if configure to do so
+    integrationret=0
+    if [ "$want_integration_test" = "1" ]; then
+        # Horizon / Dashboard is installed in the controller node
+	oncontroller_run_integration_test
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: looks like there is an indentation problem here..

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -3816,6 +3880,13 @@ function oncontroller_testsetup()
         tempestret=$?
     fi
 
+    # Run Horizon integration test if configure to do so
+    integrationret=0
+    if [ "$want_integration_test" = "1" ]; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: maybe we want to label that want_horizon_integration_test ? or do we plan to reuse that? then maybe it should be ..._tests  instead?

**Comment by aplanas** [comment]
* Fix indentation problem

**Comment by gosipyan** [comment]
One test requires network with name public .So i suggest add following command also to setup .
neutron net-create public	
neutron subnet-create public 192.168.2.0/24 --name ext-ne
Maybe subnet is no needed.

**Pull request merged**